tests/secret-providers/gcp_secret_manager/gcp_secret_manager_test.go

```diff
@@ -0,0 +1,476 @@
+//go:build e2e
```

Comment:
Could we add another test (just like this one) testing the pod identity authentication? The service account is already generated, so it should be just updating the TriggerAuthentication

Reply:
Sure, however, if I understand correctly this test has to be executed using a GKE cluster to use the pod identity authentication (via the Workload Identity), how will that be provisioned?

Reply:
No, you can execute it in our current cluster :) 
If you check gcp scalers tests, you will see that we are already testing it: https://github.com/kedacore/keda/tree/main/tests/scalers/gcp
For that, we are installing an [external component](https://github.com/kedacore/keda/blob/main/tests/utils/setup_test.go#L131-L151) for mutating KEDA pods, but it works. We do the same for AWS role assumption.

For adding the test, just copy/paste current case and enable pod identity in the TriggerAuthentication and that's all (apart from rename unique stuff, like namespace and the secret key)

Reply:
I have added the test but wasn't able to test it locally

Reply:
I'm triggering it 🤞

Reply:
it passes